frontend/src/utils/hooks.ts

```diff
@@ -1,35 +1,58 @@
+import { apiBackend } from '../api/apiClient'
 
 export const useValidPositionsAsync = (
     setValidPositions: (newPositions: PositionDetails[]) => void,
     setHasFetched: (newState: boolean) => void,
-    currentContext: Context | null
+    currentContext: Context | null,
+    errorProps: ErrorProps
 ) => {
     const apiClients = useApiClients()
     const positionNames = ['CC Manager', 'Construction Manager', 'Commissioning Manager', 'Project Manager', 'Project Director']
     useEffect(() => {
         //Filter out correct positions. map id, name and assigned person and corresponding name. Finally filter invalid elements
         if (currentContext?.externalId) {
-            apiClients.org.getPositionsAsync(currentContext.externalId).then(response => {
-                setValidPositions(
-                    response.data
-                        .filter(position => positionNames.some(substring => position.name.includes(substring)))
-                        .map(
-                            (position): PositionDetails => ({
-                                positionId: position.id,
-                                positionName: position.name,
-                                assignedPerson: getValidPosition(position.instances)?.assignedPerson,
-                                assignedPersonName: getName(getValidPosition(position.instances)),
-                            })
-                        )
-                        .filter(element => element.assignedPerson !== undefined && element.assignedPerson !== null)
-                )
-                setHasFetched(true)
-            })
+            apiClients.org.getPositionsAsync(currentContext.externalId).then(
+                response => {
+                    setValidPositions(
+                        response.data
+                            .filter(position => positionNames.some(substring => position.name.includes(substring)))
+                            .map(
+                                (position): PositionDetails => ({
+                                    positionId: position.id,
+                                    positionName: position.name,
+                                    assignedPerson: getValidPosition(position.instances)?.assignedPerson,
+                                    assignedPersonName: getName(getValidPosition(position.instances)),
+                                })
+                            )
+                            .filter(element => element.assignedPerson !== undefined && element.assignedPerson !== null)
+                    )
+                    setHasFetched(true)
+                },
+                reason => {
+                    errorProps.setErrorMessage(reason)
+                    errorProps.setIsError(true)
+                }
+            )
         }
     }, [currentContext])
 }
+
+export const getCountriesAsync = (setCountryList: (newCountries: string[]) => void, errorProps: ErrorProps) => {
```

Comment:
Maybe this function can be renamed to `useGetCountriesAsync` to better indicate that it is a hook?

Reply:
solved